Description: Prevent unaligned memory access on armhf in connect storage engine
 Make use of memcpy instead of straight pointer dereferencing. The surrounding
 code uses the same logic and should be fixed upstream.
Author: Vicen»õiu Ciorbaru <vicentiu@mariadb.org>

diff --git a/storage/connect/value.cpp b/storage/connect/value.cpp
index 64d0e13..bef035d 100644
--- a/storage/connect/value.cpp
+++ b/storage/connect/value.cpp
@@ -796,7 +796,7 @@ uchar TYPVAL<uchar>::GetTypedValue(PVBLK blk, int n)
 template <class TYPE>
 void TYPVAL<TYPE>::SetBinValue(void *p)
   {
-  Tval = *(TYPE *)p;
+  memcpy(&Tval, p, sizeof(TYPE));
   Null = false;
   } // end of SetBinValue
 
@@ -819,7 +819,7 @@ bool TYPVAL<TYPE>::GetBinValue(void *buf, int buflen, bool go)
 //#endif
 
   if (go)
-    *(TYPE *)buf = Tval;
+    memcpy(buf, &Tval, sizeof(TYPE));
 
   Null = false;
   return false;
